### ISSUE TITLE
ArgumentError: invalid value for Hash() when working with jsonb arrays

### ISSUE THREAD

**Issue by whatthewhat** [hanami/model#418]
Hi!
Looks like using automatic schema and jsonb arrays in postgres is currently not possible.

Assuming this simple setup:
```ruby
# ...create_articles.rb
Hanami::Model.migration do
  change do
    create_table :articles do
      # ...
      column :tags, 'jsonb', null: false, default: [].to_json
    end
  end
end

# ../entities/article.rb
class Article < Hanami::Entity
end

# ../repositories/article_repository.rb
class ArticleRepository < Hanami::Repository
end
```

I get a ArgumentError when trying to create a record with a jsonb array:
```ruby
repository = ArticleRepository.new
repository.create(tags: ['foo', 'bar'])
# => ArgumentError: invalid value for Hash(): ["foo", "bar"]
```

After some digging and a lot of `bundle open`_ing_ I'm pretty sure this line is to blame: https://github.com/hanami/model/blob/36003ff50c571574269cd81d0c0554cc9d949bf5/lib/hanami/model/sql/types.rb#L76, basically all jsonb columns are treated as hashes, which is what causes the exception.

The workarounds that I was able to find are:

1. Not using automatic schema (sad face)
2. This horrible hack to manually set the column type, that uses at least 2 levels of private APIs:

```ruby
class Article < Hanami::Entity
  def self.schema=(schema)
    schema.send(:schema).type.options[:member_types][:tags] = Types::Json::Array
    super
  end
end
```

_sidenote:_ I wonder if https://github.com/hanami/model/issues/385 was a final decision? Having a public API for the above seems pretty useful.

I'd be up for helping to fix this (assuming it's a bug), but I've got very limited experience with hanami and rom, so might need some guidance to do so...

**Comment by jodosha** [comment]
@whatthewhat Hi, I'm sorry about this problem. We assumed: JSONB => `Hash`, PG Array => `Array`.

Thanks to your debugging I got straight to the point and wrote a patch for this: https://github.com/hanami/model/pull/422

Can you please verify if it works for you? Thanks.

**Comment by whatthewhat** [comment]
@jodosha just tested and both #422 and #423 seem to be working, cheers! 👍

**Comment by jodosha** [comment]
Fixed by #423
